projects/aca-content/src/lib/components/search/search-save/dialog/edit/saved-search-edit-dialog.component.spec.ts

```diff
@@ -0,0 +1,109 @@
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail. Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * The Alfresco Example Content Application is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { ComponentFixture, fakeAsync, TestBed, tick } from '@angular/core/testing';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+import { of } from 'rxjs';
+import { SavedSearchEditDialogComponent } from './saved-search-edit-dialog.component';
+import { ContentTestingModule, SavedSearch, SavedSearchesService } from '@alfresco/adf-content-services';
+import { provideMockStore } from '@ngrx/store/testing';
+import { Store } from '@ngrx/store';
+import { SnackbarErrorAction, SnackbarInfoAction } from '@alfresco/aca-shared/store';
+import { AppTestingModule } from '../../../../../testing/app-testing.module';
+
+describe('SaveSearchEditDialogComponent', () => {
+  let fixture: ComponentFixture<SavedSearchEditDialogComponent>;
+  let component: SavedSearchEditDialogComponent;
+  let savedSearchesService: SavedSearchesService;
+  let store: Store;
+  let submitButton: HTMLButtonElement;
+  const savedSearchToDelete: SavedSearch = {
+    name: 'test',
+    encodedUrl: '1234',
+    order: 0
+  };
+
+  const dialogRef = {
+    close: jasmine.createSpy('close')
+  };
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [ContentTestingModule, AppTestingModule],
+      providers: [
+        { provide: MatDialogRef, useValue: dialogRef },
+        provideMockStore(),
+        { provide: SavedSearchesService, useValue: { editSavedSearch: () => of() } },
+        { provide: MAT_DIALOG_DATA, useValue: savedSearchToDelete }
+      ]
+    });
+    dialogRef.close.calls.reset();
+    fixture = TestBed.createComponent(SavedSearchEditDialogComponent);
+    component = fixture.componentInstance;
+    savedSearchesService = TestBed.inject(SavedSearchesService);
+    store = TestBed.inject(Store);
+
+    submitButton = fixture.nativeElement.querySelector('#aca-saved-search-edit-dialog-submit-button');
+  });
+
+  afterEach(() => {
+    fixture.destroy();
+  });
+
+  it('should not save search if form is invalid', fakeAsync(() => {
```

Comment:
I guess a better test case would be 'should disable submit button if form is invalid' , as that seems to be more inline with the way the component is behaving